students/chelsea_nayan/lesson03/tests/test_unit.py

```diff
@@ -0,0 +1,206 @@
+        self.assertEqual(test_customer_04.c_phone_number, self.CUSTOMER_LIST[3][4])
+        self.assertEqual(test_customer_04.c_email_address, self.CUSTOMER_LIST[3][5])
+        self.assertEqual(test_customer_04.c_status, self.CUSTOMER_LIST[3][6])
+        self.assertEqual(test_customer_04.c_credit_limit, self.CUSTOMER_LIST[3][7])
+
+        with self.assertRaises(IntegrityError):
+            add_customer(*self.CUSTOMER_LIST[6])
+
+        LOGGER.info('Finished testing add_customer function!')
+
+    def test_search_customer(self):
+        '''Testing search_customer function'''
+
+        LOGGER.info('Testing search_customer function!')
+        setup()
+
+        add_customer(*self.CUSTOMER_LIST[3]) # Danny Devito
+        expected = {'customer_id': '04',
+                    'firstname': 'Danny',
+                    'lastname': 'Devito',
+                    'home_address': '400 4th Ave W',
+                    'phone_number': '(206)444-444',
+                    'email_address': 'ddevito@email.com',
+                    'status': 'Inactive',
+                    'credit_line': 500000}
+
+        empty_dict = {}
+
+        try:
+            test_dict = search_customer('04')
+            test_none = search_customer('05')
+
+            self.assertEqual(test_dict, expected)
+            self.assertEqual(test_none, empty_dict)
+
+        except DoesNotExist:
+            assert False
+
+        with self.assertRaises(DoesNotExist):
+            search_customer(self.CUSTOMER_LIST[5])
+
+        LOGGER.info('Finished testing search_customer function!')
+
+    def test_delete_customer(self):
+        '''Testing delete_customer function'''
+
+        LOGGER.info('Testing delete_customer function!')
+        setup()
+
+        add_customer(*self.CUSTOMER_LIST[0]) # Anakin Skywalker
+        add_customer(*self.CUSTOMER_LIST[1]) # Bilbo Baggins
+        add_customer(*self.CUSTOMER_LIST[2]) # Charlie Chadmeister
+        add_customer(*self.CUSTOMER_LIST[3]) # Danny Devito
+
+        test_customer_02 = Customer.get(Customer.c_id == '02')
+        self.assertEqual(test_customer_02.c_firstname, 'Bilbo')
+        delete_customer('02')
+
+        with self.assertRaises(DoesNotExist):
+            delete_customer('09')
+
+        LOGGER.info('Finished testing delete_customer function!')
+
+    def test_update_customer_credit(self):
+        '''Testing update_customer_credit function'''
+
+        LOGGER.info('Testing update_customer_credit function!')
+        setup()
+
+        add_customer(*self.CUSTOMER_LIST[2]) # Charlie Chadmeister
+        add_customer(*self.CUSTOMER_LIST[3]) # Danny Devito
+
+        test_customer_03 = Customer.get(Customer.c_id == '03')
+        test_customer_04 = Customer.get(Customer.c_id == '04')
+
+        LOGGER.info('Customer id 03 credit limit is %i: ', self.CUSTOMER_LIST[2][7])
+        LOGGER.info('Customer id 04 credit limit is %i: ', self.CUSTOMER_LIST[3][7])
+
+        update_customer_credit('03', 3000.0)
```

Comment:
You're updating the customer's credit to 3000 but later checking for that value to be 4000, and the test is passing, which indicates the update is not working. A similar thing occurs with customer #4, where the test is failing, since the update also didn't take place.